### PR TITLE
newdevice: Basic support for HPE MSA storage

### PR DESCRIPTION
**Pull request by FTBZ**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

screen shot 2017-02-15 at 16 59 13
screen shot 2017-02-15 at 16 59 04

### CODE REVIEW

File: includes/polling/os/hpe-msa.inc.php [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+$oids = 'cpqSiProductName.0 cpqSiSysSerialNum.0';
+
+$data = snmp_get_multi($device, $oids, '-OQUs', 'CPQSINFO-MIB');
+
+$hardware = preg_replace('/"/', '', $data[0]['cpqSiProductName']);
```
Comment (murrant) [inline includes/polling/os/hpe-msa.inc.php]: What is the output for these?  I don't like the preg_replace here when `str_replace()` or `trim($sr, '"')` would likely work.

Adding the output of these to the snmprec wouldn't be a bad idea either. like this:
```
1.3.6.1.4.1.232.2.2.4.2.0|4|Product name string here
1.3.6.1.4.1.232.2.2.2.1.0|4|Serial number here (Change actual number to a similar format/length)
```

File: includes/definitions/hpe-msa.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+os: hpe-msa
```
Comment (laf) [inline includes/definitions/hpe-msa.yaml]: Is storageworks not a better os name?

File: tests/snmpsim/hpe-msa.snmprec [hunk 3]
```diff
@@ -0,0 +1,2 @@
+1.3.6.1.2.1.1.1.0|4|"HP MSA 2040 SAN"
+1.3.6.1.2.1.1.2.0|6|"DRSAKZR-123456789"
```
Comment (laf) [inline tests/snmpsim/hpe-msa.snmprec]: This should be the sysObjectId

File: includes/definitions/hpe-msa.yaml (hunk 2 above)
Comment (FTBZ) [inline includes/definitions/hpe-msa.yaml]: StorageWorks isn't used anymore by HPE, this is the MSA family now.

File: tests/snmpsim/hpe-msa.snmprec (hunk 3 above)
Comment (FTBZ) [inline tests/snmpsim/hpe-msa.snmprec]: Not sure to understand, I pushed a PR tell me if it's wrong again. The OIDs should be standardised of the real OID of the device?

**Pull request merged**